mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1088,6 +1163,192 @@ func TestClientSideEncryptionProse(t *testing.T) {
 			})
 		}
 	})
+
+	// These tests only run when 3 KMS HTTP servers and 1 KMS KMIP server are running. See specification for port numbers and necessary arguments:
+	// https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst#kms-tls-options-tests
+	mt.RunOpts("kms tls options tests", noClientOpts, func(mt *mtest.T) {
+		validKmsProviders := map[string]map[string]interface{}{
+			"aws": {
+				"accessKeyId":     awsAccessKeyID,
+				"secretAccessKey": awsSecretAccessKey,
+			},
+			"azure": {
+				"tenantId":                 azureTenantID,
+				"clientId":                 azureClientID,
+				"clientSecret":             azureClientSecret,
+				"identityPlatformEndpoint": "127.0.0.1:8002",
+			},
+			"gcp": {
+				"email":      gcpEmail,
+				"privateKey": gcpPrivateKey,
+				"endpoint":   "127.0.0.1:8002",
+			},
+			"kmip": {
+				"endpoint": "127.0.0.1:5698",
+			},
+		}
+
+		expiredKmsProviders := map[string]map[string]interface{}{
+			"aws": {
+				"accessKeyId":     awsAccessKeyID,
+				"secretAccessKey": awsSecretAccessKey,
+			},
+			"azure": {
+				"tenantId":                 azureTenantID,
+				"clientId":                 azureClientID,
+				"clientSecret":             azureClientSecret,
+				"identityPlatformEndpoint": "127.0.0.1:8000",
+			},
+			"gcp": {
+				"email":      gcpEmail,
+				"privateKey": gcpPrivateKey,
+				"endpoint":   "127.0.0.1:8000",
+			},
+			"kmip": {
+				"endpoint": "127.0.0.1:8000",
+			},
+		}
+
+		invalidKmsProviders := map[string]map[string]interface{}{
+			"aws": {
+				"accessKeyId":     awsAccessKeyID,
+				"secretAccessKey": awsSecretAccessKey,
+			},
+			"azure": {
+				"tenantId":                 azureTenantID,
+				"clientId":                 azureClientID,
+				"clientSecret":             azureClientSecret,
+				"identityPlatformEndpoint": "127.0.0.1:8001",
+			},
+			"gcp": {
+				"email":      gcpEmail,
+				"privateKey": gcpPrivateKey,
+				"endpoint":   "127.0.0.1:8001",
+			},
+			"kmip": {
+				"endpoint": "127.0.0.1:8001",
+			},
+		}
+
+		// create valid Client Encryption options without TLS options
```

Comment:
```suggestion
		// create valid Client Encryption options without a client certificate.
```

This comes from the changes in https://github.com/mongodb/specifications/commit/676a4623379b211d2e29a81c73ab5bf5d47ab708. `validClientEncryptionOptionsWithoutTLS` should be renamed to `validClientEncryptionOptionsWithoutClientCert`. TLS options should configured with the `tlsCAFile` configured for each KMS provider.

Reply:
Sounds good, done